### PR TITLE
slip39 reset device layout refactor

### PR DESCRIPTION
**Pull request by ciny**
This PR implements #409 and #456

I'm opening a draft for now. I'd still like to do something about slip39_prompt_* functions however I'm not sure if there's anything to do.

### CODE REVIEW

File: core/src/apps/management/backup_device.py [hunk 1]
```diff
@@ -5,8 +5,8 @@
 from apps.common import mnemonic, storage
 from apps.management.reset_device import (
     backup_bip39_wallet,
-    backup_group_slip39_wallet,
-    backup_slip39_wallet,
+    backup_slip39_advanced_wallet,
+    backup_slip39_basic_wallet,
```
Comment (tsusanka) [inline core/src/apps/management/backup_device.py]: Maybe let's strip the `wallet`? It seems to like there is basic and advanced wallet :). So just  `backup_slip39_advanced` and `backup_slip39_basic`?

File: core/src/apps/management/reset_device/layout.py [hunk 2]
```diff
@@ -355,13 +400,33 @@ def _split_share_into_pages(share_words):
...
+            await _show_share_words(ctx, share_words, index)
+
+            # make the user confirm words from the share
+            if await _confirm_share_words(ctx, index, share_words):
+                await _show_confirmation_success(
+                    ctx, index, num_of_shares=len(shares), slip39=True
+                )
+                break  # this share is confirmed, go to next one
+            else:
+                await _show_confirmation_failure(ctx, index)
```
Comment (tsusanka) [inline core/src/apps/management/reset_device/layout.py]: This internal while loop is very similar to the one in `bip39_show_and_confirm_mnemonic`. Is it worth it to merge these? Couldn't we have a single `show_and_confirm_shares` for both actually?

File: core/src/apps/management/reset_device/layout.py [hunk 3]
```diff
@@ -245,76 +289,67 @@ def _get_mnemonic_page(words: list):
 # TODO: smaller font or tighter rows to fit more text in
 # TODO: icons in checklist
```
Comment (tsusanka) [inline core/src/apps/management/reset_device/layout.py]: Let's remove these TODOs for now. There should be an issue, if it really bothers someone :).

File: core/src/apps/management/reset_device/layout.py (hunk 2 above)
Comment (ciny) [reply core/src/apps/management/reset_device/layout.py]: I was thinking about it but we can only unify bip39 and slip39_basic, slip39_advanced has one extra loop that would be tricky to unify. So we still end up with two functions. The actual display of the words and their confirmations is unified which I think was the main goal of #409 .

File: core/src/apps/management/backup_device.py (hunk 1 above)
Comment (ciny) [reply core/src/apps/management/backup_device.py]: done

File: core/src/apps/management/reset_device/layout.py (hunk 3 above)
Comment (ciny) [reply core/src/apps/management/reset_device/layout.py]: done

File: core/src/apps/management/reset_device/layout.py (hunk 2 above)
Comment (tsusanka) [reply core/src/apps/management/reset_device/layout.py]: Fair enough.

**Comment by tsusanka** [comment]
Please squash into a single commit and mark ready for review.

**Pull request merged**